src/js/services/forms.js

```diff
@@ -160,6 +161,17 @@ export default App.extend({
         channel.request('send', 'fetch:directory', { error: responseData, requestId });
       });
   },
+  fetchIcd({ term, requestId }) {
+    const channel = this.getChannel();
+
+    return Promise.resolve(Radio.request('entities', 'fetch:icd:byTerm', term))
+      .then(icd => {
+        channel.request('send', 'fetch:icd', { value: get(icd, ['data', 'icdCodes']), requestId });
+      })
+      .catch(({ responseData }) => {
+        channel.request('send', 'fetch:icd', { error: responseData, requestId });
+      });
```

Comment:
This catch isn't doing anything when the api request returns a `400`. 

Was thinking about doing something along these lines:

```js
return Promise.resolve(Radio.request('entities', 'fetch:icd:byTerm', term))
  .then(responseData => {
    if (responseData.errors) {
      channel.request('send', 'fetch:icd', { error: responseData.errors, requestId });
      return;
    }

    channel.request('send', 'fetch:icd', { value: get(responseData, ['data', 'icdCodes']), requestId });
  });
```

But that doesn't seem 100% correct to me. Maybe something needs to be changed in the `/entities-service/graphql.js` file? To reject the promise and trigger the `catch()`.

Reply:
I added a `.then(handleJSON)` in the `src/js/entities-service/graphql.js` file here: https://github.com/RoundingWell/care-ops-frontend/pull/1238/commits/6cf646db80a4ee0356cce8e7f093ebb27449ea9c.

Instead of doing the code snippet I commented above.

The promise is now being rejected on a `400`, so the `catch()` is triggered now.